### PR TITLE
 Issue #5941: remove spaces on the end that were connected to tests 

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6865
and Issue #5941

Windows would fail without #6865 . There are still 2 areas with spaces on the end, because they are connected to code coverage.

### CODE REVIEW

**Comment by romani** [comment]
> Windows would fail without #6865 .

please share error message.

**Comment by rnveach** [comment]
> > Windows would fail without #6865 .

> please share error message. 

Failure means the assertion would fail. Users don't run our tests, but if they did something involving newlines or text before a new line with the javadocs they could see something described in #6865 ( `\r` on attribute value ).

Linux would pass because there would be no `\r` in files. Windows would fail with the following:
````
org.junit.ComparisonFailure: Generated tree from the javadoc file should match the pre-defined tree expected:<... -> "someotherthing"[] [37:34]
|         ...> but was:<... -> "someotherthing"[\r] [37:34]
|         ...>
````
Tests would pass before this PR because we swapped any `\r\n` text with `\n`, but the bug caused `\r` to be identified as text instead of as the new line and this is why the swap didn't help any.

> spaces on the end that were connected to tests sound for me like spaces that are required for validation verification.

I was trying to find wording that said "changes to these resources means test results were affected in some way". Other PR #6864 was for files that did not affect tests at all, so I was trying to find the wording to differentiate the 2.

**Comment by rnveach** [comment]
Regression on Linux: http://rveach.no-ip.org/checkstyle/regression/reports/260/

**Comment by romani** [comment]
>Users don't run our tests, 

They do run them when they to do know how to disable tests execution.
They just do `mvn install` to build jar, as in all other maven projects.

File: src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/javadocTags/expectedTextBeforeJavadocTagsAst.txt [hunk 1]
```diff
@@ -16,22 +16,22 @@ JAVADOC -> JAVADOC [0:0]
...
+|       `--NEWLINE -> \n [3:19]
 |--LEADING_ASTERISK ->  * [4:0]
 |--WS ->   [4:2]
 |--JAVADOC_TAG -> JAVADOC_TAG [4:3]
 |   |--AUTHOR_LITERAL -> @author [4:3]
 |   |--WS ->   [4:10]
 |   `--DESCRIPTION -> DESCRIPTION [4:11]
 |       |--TEXT -> Walter White [4:11]
-|       `--NEWLINE -> \r\n [4:23]
+|       `--NEWLINE -> \n [4:23]
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/javadocTags/expectedTextBeforeJavadocTagsAst.txt]: Please explain why we lost `\r`.
It is good to have inputs in windows lineends

Comment (rnveach) [reply src/test/resources/com/puppycrawl/tools/checkstyle/grammar/javadoc/javadocTags/expectedTextBeforeJavadocTagsAst.txt]: We technically never had the `\r` to begin with.
Repo is linux but CI can be run on windows and windows converts to `\r\n`. To be able to run the same tests on both CIs, we had to hack away window's `\r\n`. We did this to both expected and actual results.
See: 
https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/AbstractTreeTestSupport.java#L55-L56
and
https://github.com/checkstyle/checkstyle/blob/c7ffe16d7e090cb5832aac65b874cf869de38d49/src/test/java/com/puppycrawl/tools/checkstyle/AbstractPathTestSupport.java#L72-L74
This file is not in `.gitattributes` so it was never being forced to `\r\n` on linux, hence me saying "We technically had the `\r` to begin with".

When I was copying over actual results to expected to pass tests, this change came in.

**Comment by romani** [comment]
@pbludov , please finalize review.

**Pull request merged**